contentcuration/contentcuration/utils/channel.py

```diff
@@ -0,0 +1,142 @@
+import functools
+import math
+import random
+import time
+
+from django.core.cache import cache
+from django.db.models import Count
+from django.db.models import Sum
+from django_cte import With
+
+from contentcuration.models import Channel
+from contentcuration.models import ContentNode
+from contentcuration.models import FileCTE
+from contentcuration.models import User
+
+CACHE_CHANNEL_KEY = "channel_metadata_{}"
+
+
+def cache_stampede(expire, beta=1):
+    """Cache decorator with cache stampede protection.
```

Comment:
Can we add a comment with a link to where we vendored it from?

Reply:
I added a link to the pdf of the paper and its explanation. I'll add it to the diskcache code too, I stole some ideas from there (mainly the decorator usage) but for the main logic I used the paper. This implementation is simpler than diskcache's.

Reply:
OK, if you didn't lean on diskcache's code that much, this is fine then!